### PR TITLE
Add interface for getting execution order @open sesame 03/29 10:16

### PR DESCRIPTION
**Pull request by jihochu**
This patch implemnts debugging feature for execution order.

Execution order for each tensor is decided while initalize and finalize
phase. To check the validity of them, getting execution order interface
is added to network graph class. It can gather both variable and
gradient of the tensors.

Signed-off-by: Jiho Chu <jiho.chu@samsung.com>

### CODE REVIEW

File: nntrainer/graph/network_graph.cpp [hunk 1]
```diff
@@ -831,6 +832,58 @@ NetworkGraph::finalizeContext(const std::shared_ptr<LayerNode> &lnode,
...
+  auto tensor_specs = init_context.getTensorsSpec();
+
+  std::map<std::string, std::vector<unsigned int>> exec_orders;
+
+  for (auto &spec : out_specs) {
+    const auto name = lnode->getName() + ":" + spec.variable_spec.name;
+    auto orders = tensor_manager->getTensorExecutionOrders(name, false);
+    exec_orders.insert({name, orders});
+    try {
+      auto orders2 = tensor_manager->getTensorExecutionOrders(name + ":grad", false);
```
Comment (SeoHyungjun) [inline nntrainer/graph/network_graph.cpp:850]: Wouldn't it be more intuitive to set orders to forward_orders and orders2 to gradient_orders?

File: nntrainer/graph/network_graph.h [hunk 2]
```diff
@@ -374,6 +374,14 @@ class NetworkGraph {
    */
   void flushCacheExcept(const unsigned int order);
 
+  /**
+   * @brief Get layer node's tenexecution orders
+   *
+   * @param order except execution order
```
Comment (DonghakPark) [inline nntrainer/graph/network_graph.h:380]: Typo Error : param ```order``` --> ```lnode```

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

**Review by baek2sm — APPROVED** [review]
LGTM

File: nntrainer/graph/network_graph.h (hunk 2 above)
Comment (jihochu) [reply nntrainer/graph/network_graph.h:380]: Nicecatch!! thanks.

File: nntrainer/graph/network_graph.cpp (hunk 1 above)
Comment (jihochu) [reply nntrainer/graph/network_graph.cpp:850]: Looks better. thanks.

**Review by DonghakPark — APPROVED** [review]
LGTM!

-> git action will be fixed (sorry for bother)

File: nntrainer/graph/network_graph.cpp [hunk 3]
```diff
@@ -842,6 +843,61 @@ NetworkGraph::finalizeContext(const std::shared_ptr<LayerNode> &lnode,
   return outputs;
 }
 
+std::map<std::string, std::vector<unsigned int>>
+NetworkGraph::getLayerExecutionOrders(const std::shared_ptr<LayerNode> &lnode) {
+  const GraphNode &gnode = *lnode.get();
+  auto init_context = lnode->getInitContext();
+  auto out_specs = init_context.getOutSpecs();
```
Comment (jijoongmoon) [inline nntrainer/graph/network_graph.cpp:850]: NNtrainer changes the init_context to runContext after initialization. does it work after initialization or it only works after compile? If it is, plz leave the comment and add the error saying it has limitation.

**Comment by jijoongmoon** [comment]
I think this golden data needs to change according to #2093

File: nntrainer/graph/network_graph.cpp (hunk 3 above)
Comment (jihochu) [reply nntrainer/graph/network_graph.cpp:850]: As we discussed in offline, I'll extract `init_context` related codes to DEBUG feature.
thanks for checking :+1:

**Comment by jihochu** [comment]
> I think this golden data needs to change according to #2093

After emerging #2093, I'll change the golden data.

File: nntrainer/graph/network_graph.h [hunk 4]
```diff
@@ -378,6 +378,17 @@ class NetworkGraph {
    */
   void flushCacheExcept(const unsigned int order);
 
+#ifdef ENABLE_TEST
```
Comment (jijoongmoon) [inline nntrainer/graph/network_graph.h:381]: Great!

**Review by jijoongmoon — APPROVED** [review]
LGTM!

**Pull request merged**